pydis_site/apps/api/serializers.py

```diff
@@ -278,16 +278,20 @@ def update(self, instance: QuerySet, validated_data: list) -> list:
         ref:https://www.django-rest-framework.org/api-guide/serializers/#customizing-multiple-update
         """
         instance_mapping = {user.id: user for user in instance}
-        data_mapping = {item['id']: item for item in validated_data}
 
         updated = []
         fields_to_update = set()
-        for user_id, data in data_mapping.items():
-            for key in data:
+        for user_data in validated_data:
+            for key in user_data:
                 fields_to_update.add(key)
-            user = instance_mapping.get(user_id)
-            user.__dict__.update(data)
-            updated.append(user)
+
+                try:
+                    user = instance_mapping[user_data["id"]]
+                except KeyError:
+                    raise ValidationError({"id": f"User with id {user_data['id']} not found."})
```

Comment:
Is there a way for this to still be a 404? Will raising `NotFound` work? Is that idiomatic?

Reply:
I can write a custom validation class to return a 404 response which can be used wherever needed or I could simply do
```python
err = ValidationError({"id": f"User with id {user_data['id']} not found."})
err.status_code = 404
raise err
```

Reply:
It's kinda complicated according to this https://github.com/encode/django-rest-framework/issues/6555